http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpFullRequest.java

```diff
@@ -56,19 +56,25 @@
      * {@link SdkHttpFullRequest#builder()}.
      */
     interface Builder extends SdkHttpRequest.Builder {
+
         /**
-         * Convenience method to set the {@link #protocol()}, {@link #host()}, {@link #port()}, and
-         * {@link #encodedPath()} from a {@link URI} object.
+         * Convenience method to set the {@link #protocol()}, {@link #host()}, {@link #port()},
+         * {@link #encodedPath()} and extracts query parameters from a {@link URI} object.
          *
          * @param uri URI containing protocol, host, port and path.
          * @return This builder for method chaining.
          */
         @Override
         default Builder uri(URI uri) {
-            return this.protocol(uri.getScheme())
+            Builder builder =  this.protocol(uri.getScheme())
                        .host(uri.getHost())
                        .port(uri.getPort())
                        .encodedPath(SdkHttpUtils.appendUri(uri.getRawPath(), encodedPath()));
+            if (uri.getRawQuery() != null) {
+                SdkHttpUtils.uriParams(uri)
+                            .forEach(this::putRawQueryParameter);
+            }
+            return builder;
```

Comment:
Same here, the existing set of query parameters set on the builder should be cleared first